wagtailvideos/views/chooser.py

```diff
@@ -1,38 +1,85 @@
 import json
 
+from django.utils.translation import ugettext as _
+
 from django.shortcuts import get_object_or_404, render
 from django.urls import reverse
-from wagtail.admin.forms import SearchForm
+# from wagtail.admin.forms import SearchForm
+from wagtail.admin.forms.search import SearchForm
 from wagtail.admin.modal_workflow import render_modal_workflow
 from wagtail.admin.utils import PermissionPolicyChecker, popular_tags_for_model
 from wagtail.core.models import Collection
 from wagtail.search import index as search_index
 from wagtail.utils.pagination import paginate
 
 from wagtailvideos.forms import get_video_form
-from wagtailvideos.models import Video
+from wagtailvideos.models import Video, get_video_model
 from wagtailvideos.permissions import permission_policy
 
 permission_checker = PermissionPolicyChecker(permission_policy)
 
 
+def get_chooser_js_data():
+    """construct context variables needed by the chooser JS"""
+    return {
+        'step': 'chooser',
+        'error_label': _("Server Error"),
+        'error_message': _("Report this error to your webmaster with the following information:"),
+        'tag_autocomplete_url': reverse('wagtailadmin_tag_autocomplete'),
+    }
+
 def get_video_json(video):
     """
     helper function: given an image, return the json to pass back to the
     image chooser panel
     """
-
-    return json.dumps({
+    return {
         'id': video.id,
         'edit_link': reverse('wagtailvideos:edit', args=(video.id,)),
         'title': video.title,
         'preview': {
             'url': video.thumbnail.url if video.thumbnail else '',
         }
-    })
+    }
+
+
+def get_chooser_context(request):
+    """Helper function to return common template context variables for the main chooser view"""
+
+    collections = Collection.objects.all()
+    if len(collections) < 2:
+        collections = None
+    else:
+        collections = Collection.order_for_display(collections)
+
+    return {
+        'searchform': SearchForm(),
+        'is_searching': False,
+        'query_string': None,
+        'will_select_format': request.GET.get('select_format'),
+        'popular_tags': popular_tags_for_model(Video),
+        'collections': collections,
+    }
+
+
+
+def paginate1(request, items, page_key=1, per_page=20):
```

Comment:
This can be moved to it's own file (`wagtailvideos/paginate.py`) and ditch the `paginate` import from wagtail